src/report/pdf.rs

```diff
@@ -0,0 +1,500 @@
+                margin: 0.1in;
+                size: 8.5in auto;
+            }
+            
+            /* Match Chrome's internal PDF generation behavior */
+            html, body {
+                font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, sans-serif;
+                font-size: 12px;
+                line-height: 1.4;
+                color: #333;
+                background: white !important;
+                margin: 0;
+                padding: 0;
+                height: auto;
+                overflow: visible;
+            }
+            
+            /* Prevent page breaks in critical elements but allow natural flow */
+            h1, h2, h3, h4, h5, h6 {
+                page-break-after: avoid;
+                break-after: avoid;
+                page-break-inside: avoid;
+                break-inside: avoid;
+            }
+            
+            /* Table handling - prevent breaking table headers but allow content to flow */
+            table {
+                border-collapse: collapse;
+                width: 100% !important;
+                font-size: 11px;
+                margin-bottom: 20px;
+            }
+            
+            thead {
+                display: table-header-group;
+                page-break-inside: avoid;
+                break-inside: avoid;
+            }
+            
+            tbody {
+                display: table-row-group;
+            }
+            
+            tr {
+                page-break-inside: avoid;
+                break-inside: avoid;
+            }
+            
+            th, td {
+                border: 1px solid #ddd !important;
+                padding: 4px 8px !important;
+                text-align: left;
+                page-break-inside: avoid;
+                break-inside: avoid;
+            }
+            
+            th {
+                background-color: #f5f5f5 !important;
+                font-weight: bold;
+            }
+            
+            /* Allow natural page breaks for content sections */
+            .metrics-section, .overview-section {
+                page-break-inside: auto;
+                break-inside: auto;
+            }
+            
+            /* Ensure charts and images scale properly */
+            img, canvas, svg {
+                max-width: 100% !important;
+                height: auto !important;
+                page-break-inside: avoid;
+                break-inside: avoid;
+            }
+        </style>
+        "#
+    .to_string();
+
+    // Insert CSS before </head> or prepend if no head tag
+    if let Some(head_end) = html_content.find("</head>") {
+        let mut result = String::with_capacity(html_content.len() + print_css.len());
+        result.push_str(&html_content[..head_end]);
+        result.push_str(&print_css);
+        result.push_str(&html_content[head_end..]);
+        result
+    } else {
+        format!("{print_css}\n{html_content}")
+    }
+}
```

Comment:
I would personally move this out of pdf.rs -- as it has nothing to do with the PDF generation itself and this would avoid a lot of `#[cfg(feature = "pdf-reports")]` lines.

Reply:
we now have print.rs for things like this